src/main/java/pl/cleankod/exchange/entrypoint/ExceptionHandlerAdvice.java

```diff
@@ -1,19 +1,47 @@
 package pl.cleankod.exchange.entrypoint;
 
+import feign.FeignException;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import pl.cleankod.exchange.core.usecase.CurrencyConversionException;
 import pl.cleankod.exchange.entrypoint.model.ApiError;
 
+import javax.validation.ConstraintViolationException;
+
 @ControllerAdvice
 public class ExceptionHandlerAdvice {
 
     @ExceptionHandler({
             CurrencyConversionException.class,
-            IllegalArgumentException.class
     })
     protected ResponseEntity<ApiError> handleBadRequest(CurrencyConversionException ex) {
         return ResponseEntity.badRequest().body(new ApiError(ex.getMessage()));
     }
+
+    @ExceptionHandler({
+            IllegalArgumentException.class
+    })
+    protected ResponseEntity<ApiError> handleIllegalArgumentException(IllegalArgumentException ex) {
+        if (ex.getMessage() != null) {
+            return ResponseEntity.badRequest().body(new ApiError(ex.getMessage()));
+        }
+
+        return ResponseEntity.badRequest().body(new ApiError("Invalid input."));
+    }
+
+    @ExceptionHandler({
+            FeignException.class
+    })
+    protected ResponseEntity<ApiError> handleBadExternalServiceRequestException(FeignException ex) {
```

Comment:
There is no test case for this.

Reply:
The exception could be removed because circuit breaker returns the default value at the moment.